modules/siddhi-core/src/main/java/org/wso2/siddhi/core/SiddhiAppRuntime.java

```diff
@@ -263,6 +264,22 @@ public void addCallback(String queryName, QueryCallback callback) {
         return query(storeQuery, null);
     }
 
+    /**
+     * This method get the storeQuery and return the corresponding output and its types.
+     *
+     * @param storeQuery this storeQuery is processed and get the output attributes.
+     * @return List of output attributes
+     */
+    public List<Attribute> getStoreQueryOutputTypes(StoreQuery storeQuery) {
```

Comment:
Better return as an array of Attribute.

Reply:
Also add the 
catch (RuntimeException e) {
             if (e instanceof SiddhiAppContextException) {
....
}